docs/diagnostics/PX1063.md

```diff
@@ -0,0 +1,117 @@
+# PX1063
+This document describes the PX1063 diagnostic.
+
+## Summary
+
+| Code   | Short Description                                                                         | Type                           | Code Fix    | 
+| ------ | ----------------------------------------------------------------------------------------- | ------------------------------ | ----------- | 
+| PX1063 | The declaration of the exception class does not contain a serialization constructor. This will cause an incorrect deserialization of the exception data and will lead to runtime errors in Acumatica ERP. | Error | Available | 
+
+## Diagnostic Description
+
+Every class that that is derived from the `System.Exception` class (including PXException-derived classes) must implement the serialization constructor.
+If the serialization constructor is not defined, it will lead to the loss of exception data during serialization or deserialization and also can lead to runtime errors, especially when Acumatica ERP is used in the cluster mode.
+
+The code fix generates the serialization constructor for the exception class. The generated constructor will be private for sealed exception classes and protected for all other classes.  The generated constructor will call the base serialization constructor. 
+If the exception class declares a new serializable data then the constructor will contain a call to the `RestoreObjectProps` static method of the `PX.Common.ReflectionSerializer` class. This class is responsible for Acumatica custom reflection-based serialization and deserialization of data. If the exception class does not declare new serializable data, the serialization constructor body will be empty.
```

Comment:
No mentioning of this case like in PX1064:

> **_Note:_** In older versions of Acumatica ERP, there is no `PX.Common.ReflectionSerializer` class because it was added as a refactoring later. The `PX.Common.PXReflectionSerializer` class will be used instead, the names of the methods are the same.

We should add it to PX1063 too